src/main/java/build/buildfarm/common/grpc/TracingMetadataUtils.java

```diff
@@ -45,9 +49,14 @@ public static RequestMetadata fromCurrentContext() {
     if (metadata == null) {
       metadata = RequestMetadata.getDefaultInstance();
     }
+
     return metadata;
   }
 
+  public static Iterable<Property> headersFromCurrentContext() {
+    return capturedHeaders;
+  }
+
```

Comment:
You must change this to perform a headers extraction from the current context - you can relatively easily accomplish this by setting a separate key on the context with an additional `withValue` for a separate key.